stock_picking_line_sequence/models/stock.py

```diff
@@ -0,0 +1,98 @@
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    # re-defines the field to change the default
+    sequence = fields.Integer("HiddenSequence", default=9999)
+
+    # displays sequence on the stock moves
+    sequence2 = fields.Integer(
+        "Sequence",
+        help="Shows the sequence in the Stock Move.",
+        related="sequence",
+        readonly=True,
+        store=True,
+    )
+
+    @api.model
+    def create(self, values):
+        move = super(StockMove, self).create(values)
+        # We do not reset the sequence if we are copying a complete picking
+        # or creating a backorder
+        if not self.env.context.get("keep_line_sequence", False):
+            move.picking_id._reset_sequence()
+        return move
+
+
+class StockMoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    def _get_aggregated_product_quantities(self, **kwargs):
+        def get_aggregated_properties(move_line=False, move=False):
+            move = move or move_line.move_id
+            uom = move.product_uom or move_line.product_uom_id
+            name = move.product_id.display_name
+            description = move.description_picking
+            if description == name or description == move.product_id.name:
+                description = False
+            product = move.product_id
+            line_key = (
+                f"{product.id}_{product.display_name}_" f'{description or ""}_{uom.id}'
+            )
+            return line_key
+
+        aggregated_move_lines = super(
+            StockMoveLine, self
+        )._get_aggregated_product_quantities(**kwargs)
+        for move_line in self:
+            line_key = get_aggregated_properties(move_line=move_line)
+            sequence2 = move_line.move_id.sequence2
+            if line_key in aggregated_move_lines:
+                aggregated_move_lines[line_key]["sequence2"] = sequence2
+
+        return aggregated_move_lines
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    @api.depends("move_ids_without_package")
+    def _compute_max_line_sequence(self):
+        """Allow to know the highest sequence entered in move lines.
+        Then we add 1 to this value for the next sequence, this value is
+        passed to the context of the o2m field in the view.
+        So when we create new move line, the sequence is automatically
+        incremented by 1. (max_sequence + 1)
+        """
+        for picking in self:
+            picking.max_line_sequence = (
+                max(picking.mapped("move_ids_without_package.sequence") or [0]) + 1
+            )
+
+    max_line_sequence = fields.Integer(
+        string="Max sequence in lines", compute="_compute_max_line_sequence"
+    )
+
+    def _reset_sequence(self):
+        for rec in self:
+            current_sequence = 1
+            for line in rec.move_ids_without_package:
+                line.sequence = current_sequence
+                current_sequence += 1
```

Comment:
As far as I checked, calling `_reset_sequence` from `create` of stock_move will get two moves (one is virtual record, the other is real record). So, that will increase +1 to the sequence for read move record. May be this can be solved but not sure it is good way or not.
```
       for line in rec.move_ids_without_package:
            # Check if the record ID is an integer (real ID) or a string (virtual ID)
            if isinstance(line.id, int):
                line.sequence = current_sequence
                current_sequence += 1
```

Reply:
Incorporated these changes.. it is fixed!